ckanext/canada/plugins.py

```diff
@@ -921,31 +1016,6 @@ def get_validators(self):
             }
 
 
-class LogExtraMiddleware(object):
-    def __init__(self, app, config):
-        self.app = app
-
-    def __call__(self, environ, start_response):
-        def _start_response(status, response_headers, exc_info=None):
-            extra = []
-            try:
-                contextual_user = g.user
-            except (TypeError, RuntimeError, AttributeError):
-                contextual_user = None
-            if contextual_user:
-                log_extra = g.log_extra if hasattr(g, 'log_extra') else ''
-                #FIXME: make sure username special chars are handled
-                # the values in the tuple HAVE to be str types.
-                extra = [('X-LogExtra', f'user={contextual_user} {log_extra}')]
-
-            return start_response(
-                status,
-                response_headers + extra,
-                exc_info)
-
-        return self.app(environ, _start_response)
```

Comment:
This will render the graphana reporting useless for APIs. Consider removing it when graphana and log reporting are no longer used

Reply:
Our graphana "fields" broke a while ago and we never got them "fixed." So for CKAN logs in Graphana, we really only have the raw log files. And I am still logging to the syslog so it helps us debug and we still have those logs going to LogAnalytics.

I still have to figure out how to purge all these DB logs though, and how often we want to do that.

Reply:
Even without LogAnalytics custom fields this lets us quickly check the slowest api calls and immediately tell which user and org were making them.

I'm just suggesting removing this code as a later PR because it's been very useful in the past.

Reply:
which part of this code logs the execution time? is it just some CKAN core thing with the log_extra? the logging here isn't enough eh? https://github.com/open-data/ckanext-canada/blob/removal/api-logging-middleware/ckanext/canada/plugins.py#L866

Reply:
nginx logs include execution time